polaris-react/src/components/Text/Text.tsx

```diff
@@ -88,6 +88,19 @@ export const Text = ({
   visuallyHidden = false,
   textDecorationLine,
 }: TextProps) => {
+  const deprecatedVariants = ['heading2xl', 'heading3xl'];
+
+  if (
+    process.env.NODE_ENV === 'development' &&
+    variant &&
+    deprecatedVariants.includes(variant)
+  ) {
+    // eslint-disable-next-line no-console
+    console.warn(
+      `Deprecation: <Text variant="${variant}" />. The value "${variant}" will be removed in a future major version of Polaris. Use "displayXl" instead.`,
```

Comment:
I'm not familiar with `displayXl`, however, I'm curious if the suggest works for all `deprecatedVariants`? Should the `deprecatedVariants` include an object with `from`/`to` properties?